app/assets/javascripts/discourse/app/components/translation-placeholder.gjs

```diff
@@ -0,0 +1,45 @@
+import Component from "@glimmer/component";
+import { get } from "@ember/helper";
+
+/**
+ * Internally used by the Translation component to render placeholder
+ * content. This component conditionally renders its content only when the
+ * placeholder name matches the expected placeholder key.
+ *
+ * This component is only used through the Translation component's yielded
+ * Placeholder component, rather than directly.
```

Comment:
To prevent people using it directly, perhaps we should move it to be inlined in the main `translation.gjs` file? With gjs, we can have 'local' components which don't need to be exported.